docs/lib/predictions/fragments/android/identify-text.md

```diff
@@ -0,0 +1,65 @@
+## Set up your backend
+
+If you haven't already done so, run `amplify init` inside your project and then `amplify add auth` (we recommend selecting the *default configuration*).
+
+Run `amplify add predictions`, then use the following answers:
+
+```console
+? Please select from one of the categories below (Use arrow keys)
+❯ Identify
+  Convert
+  Interpret
+  Infer
+  Learn More
+
+? What would you like to identify? (Use arrow keys)
+❯ Identify Text
+  Identify Entities
+  Identify Labels
+
+? Provide a friendly name for your resource
+  <Enter a friendly name here>
+
+? Would you also like to identify documents? (y/N)
+  <Enter 'y'>
+
+? Who should have access? (Use arrow keys)
+  Auth users only
+❯ Auth and Guest users
+```
+
+## Working with the API
+
+### Detect text in an image
+
+Amplify will make calls to both Amazon Textract and Rekognition depending on the type of text you are looking to identify (i.e. image or document).
+
+Passing in `TextFormatType.PLAIN` as the identify action will yield `IdentifyResult`, which must be cast into `IdentifyTextResult`. See below for an example of plain text detection from an image.
+
+```java
+public void detectText(Bitmap image) {
+    Amplify.Predictions.identify(
+            TextFormatType.PLAIN,
+            image,
+            result -> Log.i("PredictionsQuickstart", ((IdentifyTextResult) result).getFullText()),
+            error -> Log.e("PredictionsQuickstart", error.toString(), error)
+    );
+}
+```
+
+**Note**: Do *NOT* pass `IdentifyActionType.DETECT_TEXT` as identify action. Pass in an instance of `TextFormatType` instead.
```

Comment:
This is confusing to me. Why the screaming **NOT**? Is this just a helpful warning?

![](https://cdn.pastemagazine.com/www/system/images/photo_albums/waynes-world-anniversary/large/waynes-world-5.jpg?1384968217)

Reply:
Yes. Based on design, `IdentifyActionType` enum is normally what you pass here... However, it returns an error when you pass `IdentifyActionType#DETECT_TEXT`.

In iOS, `IdentifyAction` uses [enum within enum](https://github.com/aws-amplify/amplify-ios/blob/master/Amplify/Categories/Predictions/Models/IdentifyAction.swift) to allow customer to specify a specific type of detection (sub-sub-category of "identify" if you will). Unfortunately, Android can't really do that the way iOS is doing, so I wrote an interface to simulate it. The way I did it was:

[`IdentifyActionType`](https://github.com/aws-amplify/amplify-android/blob/master/core/src/main/java/com/amplifyframework/predictions/models/IdentifyActionType.java), [`TextFormatType`](https://github.com/aws-amplify/amplify-android/blob/master/core/src/main/java/com/amplifyframework/predictions/models/TextFormatType.java), and [`LabelType`](https://github.com/aws-amplify/amplify-android/blob/master/core/src/main/java/com/amplifyframework/predictions/models/LabelType.java) are enums that implement [`IdentifyAction`](https://github.com/aws-amplify/amplify-android/blob/master/core/src/main/java/com/amplifyframework/predictions/models/IdentifyAction.java):

Where `IdentifyAction` is required to be specified, customer must supply it with an instance of `IdentifyActionType` UNLESS that sub-category has sub-sub-categories, in which case the other type must be specified. In hindsight, I think it makes sense for it to also accept `IdentifyActionType` here and then just assume a default value for `LabelType` or `TextFormatType`... I think I'll do that actually...

Reply:
Okay, but I guess as a reader.. tell me what I should do instead of leading with what I shouldn't do.

Reply:
removed warning since the code snippet already tells the user to use the correct enum. In the future I'll add the code changes to make it not breaking to begin with.